nucleus/modelci/client.py

```diff
@@ -29,9 +37,54 @@ def __repr__(self):
     def __eq__(self, other):
         return self.connection == other.connection
 
+    def upload_eval_function(
+        self, eval_function: Metric, name: str
+    ) -> CustomEvalFunction:
+        """Uploads an evaluation function and validates the evaluation function. ::
+
+            import nucleus
+            from nucleus.metrics import Metric, MetricResult
+            from nucleus.annotation import AnnotationList
+            from nucleus.prediction import PredictionList
+
+            client = nucleus.NucleusClient("YOUR_SCALE_API_KEY")
+
+            class MyMetric(Metric):
+                def __call__(
+                    self, annotations: AnnotationList, predictions: PredictionList
+                ) -> MetricResult:
+                    value = (len(annotations) - len(predictions)) ** 2
+                    weight = len(annotations)
+                    return MetricResult(value, weight)
+
+            eval_func = client.modelci.upload_eval_function(MyMetric(), "my_metric")
+
+        Args:
+            eval_function: an instance of the evaluation function - must be a
+                subclass of the :class:`Metric` class.
+            name: unique name of evaluation function
+        Returns:
+            :class:`CustomEvalFunction`: A container for the uploaded eval function.
+        """
+        assert isinstance(
+            eval_function, Metric
+        ), f"Expected eval_function to be of type Metric but got {type(eval_function)}"
+        response = self.connection.post(
+            EvalFunctionInput.from_metric(eval_function, name).dict(),
+            "modelci/eval_fn",
+        )
+        print(replace_double_slashes(response[LOGS_KEY]))
+        if response["eval_fn"] is None:
+            raise UploadEvalFunctionError(
+                "Eval function upload failed. See logs for traceback."
+            )
```

Comment:
I'd create a `pydantic` model for this. I'd actually use a `pydantic` model for everything that goes over the wire since it helps with catching common formatting errors/ unexpected values.

You could actually replace the double slashes in the `pydantic` model and you could move this validation logic to a `@validator`

Reply:
Yup, loving the validation and formatting that pydantic brings!